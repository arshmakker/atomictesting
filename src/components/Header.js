import React, { useState } from 'react'
import Search from './Search'
import LogoBar from './LogoBar'
import InitialList from './InitialList'
import ResultsList from './ResultsList'


function Header({ InitialData = new Set(["1st Option",
    "2nd Option",
    "3rd Option"]) }) {
    const [results, setResults] = useState([])

    const searchTriggered = keyword => {
        let valuesFound = [];
        InitialData.forEach(ele => {
            if (ele.match(keyword)) {
                valuesFound.push(ele)
            }
        })
        setResults(new Set(valuesFound))
    }

    return (
        <>
            <div style={{
                display: "flex", margin: "20px",
                justifyContent: "space-around",
                background: "lightblue"
            }}>
                <LogoBar />
                <Search onSearchResults={searchTriggered} />
            </div>
            <div style={{
                display: "flex", margin: "20px",
                justifyContent: "space-around",
            }}>
                <div style={{
                    padding: "20px",
                    background: "lightgrey"
                }}>
                    <InitialList list={InitialData} />
                </div>
                <div style={{
                    padding: "20px",
                    background: "lightgreen"
                }}>
                    <ResultsList list={results} />
                </div>
            </div>
        </>
    )
}

export default Header