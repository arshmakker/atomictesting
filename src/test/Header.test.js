import React from 'react'
import Header from '../components/Header'
import renderer from 'react-test-renderer'

const classHeader = "sample class"
const SampleData = new Set(["1st Sample",
    "2nd Sample",
    "3rd Sample"])

it("Renders the Header correctly with no props ", () => {
    const tree = renderer
        .create(<Header />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

it("Renders the Header correctly with test props", () => {
    const tree = renderer
        .create(<Header InitialData={SampleData}
            classHeader={classHeader}>
        </Header>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})