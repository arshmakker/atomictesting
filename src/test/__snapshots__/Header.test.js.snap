// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Header correctly with no props  1`] = `
<div
  style={
    Object {
      "background": "lightblue",
      "display": "flex",
      "justifyContent": "space-around",
      "margin": "20px",
    }
  }
>
  <div>
    Ma Web
  </div>
  <img
    alt="unapologetic logo"
    className=""
    src="https://img.icons8.com/ios/50/000000/pick-up.png"
  />
  <div
    className=""
  >
    <button
      className=""
    >
      Search
    </button>
    <input
      id="searchText"
      placeholder="Type here to search"
      type="text"
    />
  </div>
</div>
`;

exports[`Renders the Header correctly with test props 1`] = `
<div
  style={
    Object {
      "background": "lightblue",
      "display": "flex",
      "justifyContent": "space-around",
      "margin": "20px",
    }
  }
>
  <div>
    Ma Web
  </div>
  <img
    alt="unapologetic logo"
    className=""
    src="https://img.icons8.com/ios/50/000000/pick-up.png"
  />
  <div
    className=""
  >
    <button
      className=""
    >
      Search
    </button>
    <input
      id="searchText"
      placeholder="Type here to search"
      type="text"
    />
  </div>
</div>
`;
